app/controllers/application_controller.rb

```diff
@@ -1,13 +1,26 @@
+require 'doorkeeper_auth'
+
 class ApplicationController < ActionController::Base
+  include DoorkeeperAuth
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   force_ssl if: :force_ssl?
 
+  api_accessible! false
+
   include CurrentUser # must be after protect_from_forgery, so that authenticate! is called
 
+  rescue_from(DoorkeeperAuth::DisallowedAccessError) do |message|
+    if Rails.env.test?
```

Comment:
actually, why is the test environment special? You're specifically not rescuing in test, so the error is being raised. That's what I would expect.